Remove rpc fork and dist autograd fork tests from PyTorch repo

zhaojuanmao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29827 Remove rpc fork and dist autograd fork tests from PyTorch repo**

There are known issues for "fork tests + OMP" in Pytorch, rpc and dist autograd tests use OMP thread pools, this caused rpc fork and dist autograd fork tests to be flaky. So remove these fork tests from PyTorch repo. rpc spawn and dist autograd spawn tests are still running.

Differential Revision: [D18507384](https://our.internmc.facebook.com/intern/diff/D18507384/)

zhaojuanmao committed:
Remove rpc fork and dist autograd fork tests from PyTorch repo

There are known issues for "fork tests + OMP" in Pytorch, rpc and dist autograd tests use OMP thread pools, this caused rpc fork and dist autograd fork tests to be flaky. So remove these fork tests from PyTorch repo. rpc spawn and dist autograd spawn tests are still running.

Differential Revision: [D18507384](https://our.internmc.facebook.com/intern/diff/D18507384/)

[ghstack-poisoned]

xush6528 commented:
Can you do the same to `test_dist_optimizer.py` as well. It also depends on ProcessGroupAgent.

xush6528 commented on test/run_test.py:
```diff
@@ -64,13 +64,11 @@
     TESTS.extend([
-        'rpc_fork',
         'rpc_spawn',
-        'dist_autograd_fork',
```
Can you do the same to `test_dist_optimizer_fork.py`. It depends on `RpcAgent` as well.

zhaojuanmao replied:
will do

mrshenli commented on test/run_test.py:
```diff
@@ -64,13 +64,11 @@
 if PY33:
     TESTS.extend([
-        'rpc_fork',
         'rpc_spawn',
```
since we only have spawn mode on the OSS side, do we still need to explicitly have a "spawn" in the file name?

zhaojuanmao replied:
we can keep it for now? once we completely get rid of fork, we can merge rpcSpawnTest with RpcTest

zhaojuanmao committed:
Update on "Remove rpc fork and dist autograd fork tests from PyTorch repo"


There are known issues for "fork tests + OMP" in Pytorch, rpc and dist autograd tests use OMP thread pools, this caused rpc fork and dist autograd fork tests to be flaky. So remove these fork tests from PyTorch repo. rpc spawn and dist autograd spawn tests are still running.

Differential Revision: [D18507384](https://our.internmc.facebook.com/intern/diff/D18507384/)

[ghstack-poisoned]

mrshenli reviewed (approved):
lint and test failures are irrelevant. 

Curious, why dist_optimizer_spawn is not in windows and rocm blacklist?

zhaojuanmao commented:
I just saw @xush6528 added dist_optim_spawn to blacklist in #29747

zhaojuanmao committed:
Update on "Remove rpc fork and dist autograd fork tests from PyTorch repo"


There are known issues for "fork tests + OMP" in Pytorch, rpc and dist autograd tests use OMP thread pools, this caused rpc fork and dist autograd fork tests to be flaky. So remove these fork tests from PyTorch repo. rpc spawn and dist autograd spawn tests are still running.

Differential Revision: [D18507384](https://our.internmc.facebook.com/intern/diff/D18507384/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "Remove rpc fork and dist autograd fork tests from PyTorch repo"


There are known issues for "fork tests + OMP" in Pytorch, rpc and dist autograd tests use OMP thread pools, this caused rpc fork and dist autograd fork tests to be flaky. So remove these fork tests from PyTorch repo. rpc spawn and dist autograd spawn tests are still running.

Differential Revision: [D18507384](https://our.internmc.facebook.com/intern/diff/D18507384/)

[ghstack-poisoned]

zhaojuanmao commented:
checked "rpc_fork", "dist_autograd_fork" and "dist_optimizer_fork" did not run but "rpc_spawn", "dist_autograd_spawn" and "dist_optimizer_spawn" ran in https://circleci.com/api/v1.1/project/github/pytorch/pytorch/3645637/output/105/0?file=true&allocation-id=5dd30d9f0d6a0b62d1d857f9-0-build%2F10D9680D

zhaojuanmao commented:
rocm test failure is not relevant 

FAILED (skipped=12, unexpected successes=1)
21:57:37 Traceback (most recent call last):
21:57:37   File "test/run_test.py", line 448, in <module>
21:57:37     main()
21:57:37   File "test/run_test.py", line 441, in main
21:57:37     raise RuntimeError(message)
21:57:37 RuntimeError: test_cuda failed!
21:57:37 + cleanup
21:57:37 + retcode=1
21:57:37 + set +x
